logging + DHCP.validateReq()

pbertera opened the pull request:
Sorry I messed up my git repo, here a new PR with both patches.

psychomario commented:
The children logging levels are being set twice. First where the children are created (`tftp_logger = sys_logger.getChild("TFTP")` `pypxe-server.py` L123), and secondly inside the respective classes (`if self.mode_debug:` `tftp.py` L39). This only needs to be done in one of the two places although I don't mind which.

Also can you clean up the commits and resolve the conflicts so I can cleanly merge this?

Thanks

mmattioli commented:
@pbertera you need to do a rebase otherwise this can't be merged cleanly. Rebase off of `psychomario:development` and recommit your work on the logger. There are a few minor tweaks I will point out momentarily so you don't have to go back and do them twice but a rebase is necessary not only for this PR to be merged cleanly but for future look back on development, there needs to be linear progression.

mmattioli commented on pypxe-server.py:
```diff
@@ -48,9 +49,11 @@
         parser.add_argument('--http', action = 'store_true', dest = 'USE_HTTP', help = 'Enable built-in HTTP server', default = False)
         parser.add_argument('--no-tftp', action = 'store_false', dest = 'USE_TFTP', help = 'Disable built-in TFTP server, by default it is enabled', default = True)
-        parser.add_argument('--debug', action = 'store', dest = 'MODE_DEBUG', help = 'Comma Seperated (http,tftp,dhcp). Adds verbosity to the selected services while they run', default = '')
+        parser.add_argument('--debug', action = 'store', dest = 'MODE_DEBUG', help = 'Comma Seperated (sys,http,tftp,dhcp). Adds verbosity to the selected services while they run. Use \'all\' for enabling debug on all services', default = '')
```
I believe you are adding in `sys` because you assume the user might want to set the debug level of the pypxe server itself? We never print debug messages in `pypxe-server.py` so this should be removed unless I'm misunderstanding it's purpose.

pbertera replied:
Yes, this is the purpose, not really useful now but can be useful in the future.

mmattioli replied:
I would remove it for now only because I'm worried about confusing the user. We could explain it in the documentation but then if someone turns it on and is expecting a level of verbosity that, while supported, isn't there it could cause issues/confusion.

mmattioli commented on pypxe/dhcp.py:
```diff
@@ -39,28 +40,39 @@ def __init__(self, **serverSettings):
-            print '\tUsing HTTP Server: {}'.format(self.http)
+        self.logger.debug('NOTICE: DHCP server started in debug mode. DHCP server is using the following:')
+        self.logger.debug('  DHCP Server IP: {}'.format(self.ip))
+        self.logger.debug(' DHCP Server Port: {}'.format(self.port))
```
Just missing a whitespace here

mmattioli commented:
@psychomario good catch on the level setting. I think it should be set in the respective class rather than in `pypxe-server.py` upon creation because then if importing the respective class into a library it can't be properly set. I believe @pbertera 's thinking was that if someone imported this and wanted to use the class's logger then there should be a way to set that class's logger's debug mode. Changing the log level within the class (like in `tftp.py` L39) cleans it up and makes it work both ways.

pbertera commented:
@mmattioli yes, my thought about setting log level was with "library usage" in mind. We can remove the service setLevel() from `pypxe-server.py`, but service logger should used only inside the service and not in `pypxe-server.py`

pbertera commented:
@mmattioli I'm not a git expert, can you give me some hints about the rebase ? Thanks a lot

mmattioli commented:
@pbertera no worries :) First, backup all of your work elsewhere just incase. [This link](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) should be able to explain it better than I can type it here. After you've added the remote it should be something like:

```
git checkout development-logging
git rebase upstream/development
```

That will rebase your branch off of the development branch and then a `git status` should show you the changes that you've made and finally do a `git push` onto your feature branch (should be `origin/development-logging`). You may have to do a `git push --force` but I'm not 100% sure.

pbertera commented:
@mmattioli Thanks for the tip. This last commit should be mergeable, I removed the sys selector and fixed the missing space. I also removed the debug setter from `pypxe-server.py`

mmattioli commented on pypxe/tftp.py:
```diff
@@ -140,7 +148,6 @@ def listen(self):
             #Resent those that have timed out
             for i in self.ongoing:
                 if self.ongoing[i]['timeout']+5 < time.time() and self.ongoing[i]['retries']:
-                    print self.ongoing[i]['handle'].tell()
```
@pbertera this is that line you caught last time that @psychomario said could be removed, right?

psychomario replied:
I can confirm this.

mmattioli replied:
:+1:

mmattioli commented:
@pbertera thanks! If you could just confirm that line note above I'll merge this. Thanks again.
